In [25]:
import pandas as pd
import numpy as np

In [26]:
data = pd.read_csv('My_data.csv')

In [27]:
data

Job titiles AI Impact  Tasks  AI models  AI_Workload_Ratio  \
0     Communications Manager       98%    365       2546           0.143362   
1             Data Collector       95%    299       2148           0.139199   
2                 Data Entry       95%    325       2278           0.142669   
3                 Mail Clerk       95%    193       1366           0.141288   
4         Compliance Officer       92%    194       1369           0.141709   
...                      ...       ...    ...        ...                ...   
4701                  Singer        5%    686       2798           0.245175   
4702                 Airport        5%    556       2206           0.252040   
4703                Director        5%   1316       4695           0.280298   
4704                   Nurse        5%    710       2594           0.273709   
4705              Technician        5%    825       3256           0.253378   

                         Domain  
0            Communication & PR  
1                     Data & IT  
2     Administrative & Clerical  
3         Leadership & Strategy  
4          Medical & Healthcare  
...                         ...  
4701                  Data & IT  
4702  Administrative & Clerical  
4703      Leadership & Strategy  
4704       Medical & Healthcare  
4705   Supply Chain & Logistics  

[4706 rows x 6 columns]

The dataframe contains 4706 entries and 6 variables

Job titles variables is misspelled and needs to be fixed

It seems that some job titles may not fit certain domains, such as in row 4701 in the Data & IT domain we find a Singer job title. For this analysis this will be ignored due to it needing manual search of each job in every domain and knowledge of each undustry to rate the relevance of the job title for each domain.

In [28]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4706 entries, 0 to 4705
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Job titiles        4706 non-null   object 
 1   AI Impact          4706 non-null   object 
 2   Tasks              4706 non-null   int64  
 3   AI models          4706 non-null   int64  
 4   AI_Workload_Ratio  4706 non-null   float64
 5   Domain             4706 non-null   object 
dtypes: float64(1), int64(2), object(3)
memory usage: 220.7+ KB


All the data looks good and There are no missing values

In [29]:
# Assigning columns to rename
column_rename_mapping = {
    'Job titiles': 'Job Title'
}

# Applying column renaming to dataframe
data.rename(columns=column_rename_mapping, inplace=True)

In [30]:
unique_domains = data['Domain'].unique()

print('All the domains available:')
print(unique_domains)

All the domains available:
['Communication & PR' 'Data & IT' 'Administrative & Clerical'
 'Leadership & Strategy' 'Medical & Healthcare' 'Supply Chain & Logistics'
 'Law Enforcement' 'Construction' 'Sales & Marketing' 'Hospitality']


In [31]:
Domain_categories = data.groupby('Domain')['Job Title'].count()

# Converting results to a dictionary
result = Domain_categories.to_dict()

# Displaying results
for category, offense_types in result.items():
    print(f'Domain Category: {category}')
    print(f'Number of jobs present in the domain: {offense_types}')
    print()

Domain Category: Administrative & Clerical
Number of jobs present in the domain: 471

Domain Category: Communication & PR
Number of jobs present in the domain: 471

Domain Category: Construction
Number of jobs present in the domain: 470

Domain Category: Data & IT
Number of jobs present in the domain: 471

Domain Category: Hospitality
Number of jobs present in the domain: 470

Domain Category: Law Enforcement
Number of jobs present in the domain: 470

Domain Category: Leadership & Strategy
Number of jobs present in the domain: 471

Domain Category: Medical & Healthcare
Number of jobs present in the domain: 471

Domain Category: Sales & Marketing
Number of jobs present in the domain: 470

Domain Category: Supply Chain & Logistics
Number of jobs present in the domain: 471



Due to the similar count of jobs in every domain, a suspicion arised in whether the domains have the same job titles.

We will be checking how many jobs the domains have in common

In [35]:
# Grouping the domains with the count of unique values found in each domain
domain_counts = data.groupby('Domain')['Job Title'].nunique().reset_index()

# Renaming 'Job Title' column for clarity
domain_counts.rename(columns={'Job Title': 'Unique Job Titles'}, inplace=True)

print(domain_counts)

                      Domain  Unique Job Titles
0  Administrative & Clerical                471
1         Communication & PR                471
2               Construction                470
3                  Data & IT                471
4                Hospitality                470
5            Law Enforcement                470
6      Leadership & Strategy                471
7       Medical & Healthcare                471
8          Sales & Marketing                470
9   Supply Chain & Logistics                471


The count of unique job titles matches th count of jobs found in the earlier cell.

We can conclude that we dont have duplicate job titles in this dataframe

In [36]:
# Save as a csv file
data.to_csv('updated_AI_data.csv')